packages/babel-helper-evaluate-path/src/index.js

```diff
@@ -1,4 +1,37 @@
+"use strict";
+
 module.exports = function evaluate(path) {
+  if (path.isReferencedIdentifier()) {
+    return evaluateIdentifier(path);
+  }
+
+  const state = {
+    confident: true
+  };
+
+  // prepare
+  path.traverse({
+    Scope(scopePath) {
+      scopePath.skip();
+    },
+    ReferencedIdentifier(idPath) {
+      const binding = idPath.scope.getBinding(idPath.node.name);
```

Comment:
This is required as well. For babel's evaluate to take care of `globals`. Here we ignore globals. In evaluatePath we depot it. If a referencedId makes it to evaluateId, we have to deopt instead of simply ignoring it. Though that code looks like it's never reached, it will be useful for detecting side-effects from other transformations.